kairon/shared/channels/broadcast/whatsapp.py

```
@@ -49,25 +51,135 @@ def send(self, recipients: List, **kwargs):
+
+        for chunk in chunked(message_list, batch_size):
+            tasks = []
+            if is_resend:
+                tasks = [self.send_template_message_retry(*msg) for msg in chunk]
+            else:
+                tasks = [self.send_template_message(*msg) for msg in chunk]
+            async def run_async(tasks_to_run):
+                failed_to_send_recipients = []
+                results = await asyncio.gather(*tasks_to_run)
+                fails = [not result[0] for result in results]
+                for msg, fail in zip(chunk, fails):
```

Ruff (0.8.2)

Function definition does not bind loop variable `chunk`

(B023)